packages/vc-verification/src/issuer-verification.ts

```diff
@@ -0,0 +1,216 @@
+        credential,
+        verifyCredentialProofCallback
+      );
+    } else if (credential) {
+      return this.verifyChainOfTrustByRoleDefinition(credential);
+    } else {
+      throw new Error('Not sufficient information');
+    }
+  }
+
+  /**
+   * Verifies chain of trust for a given holder's DID and role
+   * @param {string} credential
+   */
+  async verifyChainOfTrustByRoleDefinition(credential: IVerifiableCredential) {
+    let isVcVerified = false;
+    let hasParent = true;
+    let subjectDID = credential.credentialSubject.id;
+    let role = await this.parseRoleFromCredential(credential);
+    while (hasParent) {
+      let offChainClaim = await this._credentialResolver.getCredential(
+        subjectDID,
+        role
+      );
+      if (!offChainClaim) {
+        throw new Error('No credential found');
+      } else {
+        let issuerDID;
+        if (offChainClaim.issuedToken) {
+          issuerDID = await this.verifyIssuedToken(offChainClaim.issuedToken);
+        }
+        if (issuerDID) {
+          if (await this.verifyIssuerAuthority(role, issuerDID)) {
+            subjectDID = issuerDID;
+            if (await this.isRoleIssuerDID(role)) {
+              isVcVerified = true;
+              hasParent = false;
+            } else {
+              const issuers = await this._issuerDefResolver.getIssuerDefinition(
+                role
+              );
+              if (issuers && issuers.roleName) {
+                role = issuers.roleName;
+              }
+            }
+          } else {
+            throw new Error('Issuer is not allowed to issue credential');
+          }
+        } else {
+          throw new Error('The credential is invalid');
+        }
+      }
+    }
+    return isVcVerified;
+  }
+
+  /**
+   * Verifies credential and chain of trust with callback function
+   *
+   * TO BE COMPLETED
+   */
+  async verifyChainOfTrustCallback(
+    credential: IVerifiableCredential,
+    verifyCredentialProofCallback: (
+      credential: IVerifiableCredential
+    ) => Promise<VerificationResult>
+  ) {
+    let hasParent = true;
+    let didMatched = false;
+    while (hasParent) {
+      const role = await this.parseRoleFromCredential(credential);
+      const issuers = await this._issuerDefResolver.getIssuerDefinition(role);
+      if (issuers && issuers.did && issuers.did.length > 0) {
+        for (let i = 0; i < issuers.did.length; i++) {
+          if (issuers.did[i] == credential.issuer) {
+            didMatched = true;
+            hasParent = false;
+            break;
+          }
+        }
+      } else {
+        const issuerCredential = await this._credentialResolver.getCredential(
+          credential.issuer,
+          role
+        );
+        // @ts-ignore
+        const result = await verifyCredentialProofCallback(issuerCredential);
+        if (result) {
+          // @ts-ignore
+          credential = issuerCredential;
+        } else {
```

Comment:
Could you explain why `// @ts-ignore` is needed here?
Second thing, assigning something to `credential` is making side effects for this method, I dont think that is intended. Lets imagine this example 
```
const credential = {...} // some credential
verifyChainOfTrustCallback(credential, cb)
// at this moment credential could have different value
doSmthElse(crednetial) // Could throw an error
```

Reply:
This is an incomplete implementation, mentioned in the description as well. Also it is unclear that what interface or type of object the callback function would take while iterating over the chain of trust. So, just to move ahead with the development it was put there.

Reply:
fair enough